pysph/sph/swe/basic.py

```diff
@@ -0,0 +1,1677 @@
+            x_edge_pa = edge_pa_pos[0] + np.repeat(x_parent, n)
+            y_edge_pa = edge_pa_pos[1] + np.repeat(y_parent, n)
+
+            rho0_edge_pa = np.repeat(rho0_parent, n)
+            rho_edge_pa = np.repeat(rho_parent, n)
+            alpha_edge_pa = np.repeat(alpha_parent, n)
+            parent_idx_edge_pa = np.repeat(self.idx_pa_to_split, n)
+
+            # Note:
+            # The center daughter particle properties are set at index of
+            # parent. The properties of parent needed for further calculations
+            # are not changed for now
+
+            # Center daughter particle properties update
+            for idx in self.idx_pa_to_split:
+                self.pa_arr.m[idx] *= self.center_pa_mass_frac
+                self.pa_arr.h[idx] *= self.pa_h_ratio
+                self.pa_arr.h0[idx] *= self.pa_h_ratio
+                self.pa_arr.parent_idx[idx] = int(idx)
+
+            # Update particle array to include edge daughter particles
+            self._add_edge_pa_prop(h0_edge_pa, h_edge_pa, m_edge_pa, x_edge_pa,
+                                   y_edge_pa, rho0_edge_pa, rho_edge_pa,
+                                   u_prev_step_edge_pa, v_prev_step_edge_pa,
+                                   alpha_edge_pa, parent_idx_edge_pa)
+
+    def _get_idx_of_particles_to_split(self):
+        idx_pa_to_split = []
+        for idx, val in enumerate(self.pa_arr.pa_to_split):
+            if val:
+                idx_pa_to_split.append(idx)
+        return np.array(idx_pa_to_split)
+
+    def _get_edge_pa_positions(self, h_parent, u_parent, v_parent):
+        # Number of particles to split
+        num_of_pa_to_split = len(self.idx_pa_to_split)
+
+        n = self.num_edge_pa_after_single_split
+        theta_edge_pa = zeros(n)
+        r = self.center_and_edge_pa_separation_frac
+
+        for i, theta in enumerate(range(0, 360, 60)):
+            theta_edge_pa[i] = (pi/180)*theta
+
+        # Angle of velocity vector with horizontal
+        angle_vel = np.where(
+            (np.abs(u_parent) > 1e-3) | (np.abs(v_parent) > 1e-3),
+            np.arctan2(v_parent, u_parent), 0
+            )
+
+        # Rotates the hexagon such that its horizontal axis aligns with the
+        # direction of velocity vector
+        angle_actual = (np.tile(theta_edge_pa, num_of_pa_to_split)
+                        + np.repeat(angle_vel, n))
+
+        x = r * cos(angle_actual) * np.repeat(h_parent, n)
+        y = r * sin(angle_actual) * np.repeat(h_parent, n)
+        return x.copy(), y.copy()
+
+    def _add_edge_pa_prop(self, h0_edge_pa, h_edge_pa, m_edge_pa, x_edge_pa,
+                          y_edge_pa, rho0_edge_pa, rho_edge_pa,
+                          u_prev_step_edge_pa, v_prev_step_edge_pa,
+                          alpha_edge_pa, parent_idx_edge_pa):
+        add_prop = {}
+        for prop in self.props:
+            if prop == 'm':
+                add_prop[prop] = m_edge_pa
```

Comment:
You can do this with a dictionary and looping instead of this messy and long list of conditions.